test/client.js

```diff
@@ -57,6 +57,22 @@ describe('ServiceClient', () => {
         });
     });
 
+    it('should not throw error for empty response body', () => {
```

Comment:
This test doesn't check the behavior you want and passes regardless of the changes done in implementation.